src/Mondrian/MondrianEditor.ts

```diff
@@ -0,0 +1,105 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as vscode from 'vscode';
+import {getNonce} from '../Config/GetNonce';
+
+export class MondrianEditorProvider implements vscode.CustomTextEditorProvider {
+  public static register(context: vscode.ExtensionContext): vscode.Disposable {
+    const provider = new MondrianEditorProvider(context);
+    const providerRegistration =
+        vscode.window.registerCustomEditorProvider(MondrianEditorProvider.viewType, provider);
+    return providerRegistration;
+  }
+
+  private static readonly viewType = 'onevscode.mondrianViewer';
+
+  constructor(private readonly context: vscode.ExtensionContext) {}
+
+  /**
+   * Called when custom editor is opened.
+   */
+  public async resolveCustomTextEditor(
+      document: vscode.TextDocument, webviewPanel: vscode.WebviewPanel,
+      _token: vscode.CancellationToken): Promise<void> {
+    webviewPanel.webview.options = {
+      enableScripts: true,
+    };
+    webviewPanel.webview.html = this.getHtmlForWebview(webviewPanel.webview);
+
+    webviewPanel.webview.postMessage({
+      type: 'update',
+      text: document.getText(),
+    });
+  }
+
+  /**
+   * Get the static html used for the editor webviews.
+   */
+  private getHtmlForWebview(webview: vscode.Webview): string {
+    const prefix = 'media/Mondrian';
+    const nonce = getNonce();
+
+    const scriptUri = webview.asWebviewUri(
+        vscode.Uri.joinPath(this.context.extensionUri, prefix, 'mondrianViewer.js'));
+
+    const styleUri =
+        webview.asWebviewUri(vscode.Uri.joinPath(this.context.extensionUri, prefix, 'style.css'));
+
+    return /* html */ `
+      <!DOCTYPE html>
+      <html lang="en">
+      <head>
+        <meta charset="UTF-8">
+        <meta http-equiv="Content-Security-Policy" content="default-src 'none'; img-src ${
+        webview.cspSource} data:;
+          style-src ${webview.cspSource}; script-src 'nonce-${nonce}';" />
+        <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+        <link href="${styleUri}" rel="stylesheet" />
+        <title>Mondrian Viewer</title>
+      </head>
+      <body>
+        <div class="mondrian-layout">
+          <div class="mondrian-scrollbar"></div>
+          <div class="mondrian-viewer-area">
+            <div class="mondrian-viewer-bounds"></div>
+          </div>
+          <div class="mondrian-statusbar">
+            <div class="mondrian-statusline"></div>
+            <div class="mondrian-info">
+              <b>Memory:</b> <span class="mondrian-info-memory-size">0</span> |
+              <b>Cycles:</b> <span class="mondrian-info-cycle-count">0</span> |
+              <b>Segment:</b> <select class="mondrian-segment-picker"></select>
```

Comment:
Just FYI, there is `vscode-dropdown` for `select` in VSCode :D

Reply:
`vscode-dropdown` looks nice! I've added vscode toolkit dependency and replaced `<select>` with `<vscode-dropdown>` :)